sw/device/tests/entropy_src_kat_test.c

```diff
@@ -50,6 +52,49 @@ static void entropy_with_fw_override_enable(dif_entropy_src_t *entropy_src) {
       dif_entropy_src_configure(entropy_src, config, kDifToggleEnabled));
 }
 
+/**
+ * Cleanly disables the SHA3 conditioner while in SHA3 mode, prompting
+ * the release of a conditioned seed.
+ *
+ * If stopping the conditioner fails, due to pending data keep trying for
+ * at most KAT_TEST_TIMEOUT_ATTEMPTS.
+ *
+ * @param entropy An entropy source instance.
+ */
+
+static void stop_sha3_conditioner(dif_entropy_src_t *entropy_src) {
+  uint32_t fail_count = 0;
+  dif_result_t op_result;
+
+  do {
+    op_result = dif_entropy_src_conditioner_stop(entropy_src);
+    if (op_result == kDifIpFifoFull) {
+      fail_count++;
+      CHECK(fail_count < KAT_TEST_TIMEOUT_ATTEMPTS);
+    } else {
+      CHECK_DIF_OK(op_result);
+    }
+  } while (op_result == kDifIpFifoFull);
+}
+
+/**
+ * Flushes any previously absorbed entropy from the SHA3 conditioning block.
+ *
+ * @param entropy An entropy source instance.
+ */
+
+static void flush_sha3_conditioner(dif_entropy_src_t *entropy_src) {
+  // Start and stop the conditioner, without adding any new entropy.
+  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy_src));
```

Comment:
i have a really dumb question. If the conditioner were already started, does this still work..? 
as in, do you need a disable -> enable -> disable sequence? or does enable -> enable -> disable also work?

Reply:
No, it will not, though my experience says it will be detected by the DIF... I made that mistake already.

Reply:
The real important part here is that somebody needs to send a "SHA stop" Event.   Once could in principle do "If not started: start.  Now: stop" Does that make sense?

Reply:
ah sounds good.